packages/contracts/src/dollar/core/CreditNft.sol

```diff
@@ -6,33 +6,42 @@ import "solidity-linked-list/contracts/StructuredLinkedList.sol";
 import {ICreditNft} from "../../dollar/interfaces/ICreditNft.sol";
 import "../libraries/Constants.sol";
 
-/// @title A CreditNft redeemable for dollars with an expiry block number
-/// @notice An ERC1155 where the token ID is the expiry block number
-/// @dev Implements ERC1155 so receiving contracts must implement IERC1155Receiver
+/**
+ * @notice CreditNft redeemable for Dollars with an expiry block number
+ * @notice ERC1155 where the token ID is the expiry block number
+ * @dev Implements ERC1155 so receiving contracts must implement `IERC1155Receiver`
+ * @dev 1 Credit NFT = 1 whole Ubiquity Dollar, not 1 wei
```

Comment:
I guess this comment is related to https://t.me/UbiquityDAO/30086/30794

Reply:
Exactly

I haven't tested it thoroughly yet, but from what I understand, there are some places in the code where CreditNft to Credit/Dollar conversion is wrong

Take this [method](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/src/dollar/facets/CreditNftManagerFacet.sol#L119) for example. As far as I understand, it will convert 1 whole CreditNft (which is supposed to be 1 full Dollar) to `1` wei Credit instead of `1000000000000000000` wei Credit

I will create issues later